packages/yourloops/test/integration/pages/menu/add-team-for-patient.spec.tsx

```diff
@@ -0,0 +1,191 @@
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import TeamAPI from '../../../../lib/team/team.api'
+import { getCodeTeam, teamOne } from '../../mock/team.api.mock'
+import { mockPatientLogin } from '../../mock/patient-login.mock'
+import { monitoredPatientAsTeamMember } from '../../mock/patient.api.mock'
+import { act, screen, within } from '@testing-library/react'
+import { renderPage } from '../../utils/render'
+import userEvent from '@testing-library/user-event'
+
+describe('Team menu for patient', () => {
+  beforeAll(() => {
+    mockPatientLogin(monitoredPatientAsTeamMember)
+    jest.spyOn(TeamAPI, 'getTeams').mockResolvedValue([teamOne])
+    jest.spyOn(TeamAPI, 'joinTeam').mockResolvedValue()
+  })
+  const mockGetTeamFromCod = () => {
+    jest.spyOn(TeamAPI, 'getTeamFromCode').mockResolvedValue(getCodeTeam)
+  }
+
+  it('should close the dialog after clicking the add care button with success message', async () => {
+    await act(async () => {
+      renderPage('/')
+    })
+    mockGetTeamFromCod()
+
+    const badgeTeamMenu = screen.getByLabelText('Open team menu')
+    expect(badgeTeamMenu).toHaveTextContent('1')
+    await userEvent.click(badgeTeamMenu)
+    const buttonJoinTeam = screen.getByTestId('team-menu-teams-link')
+    await userEvent.click(buttonJoinTeam)
+    const dialog = screen.getByRole('dialog')
+    const titleDialog = within(dialog).getByTestId('team-add-dialog-title')
+    const labelDialog = within(dialog).getByTestId('label-dialog')
+    expect(dialog).toBeVisible()
+    expect(titleDialog).toHaveTextContent('Join the care team')
+    expect(labelDialog).toHaveTextContent('Please enter the team code (9 digits)')
+    const cancelButton = within(dialog).getByRole('button', { name: 'Cancel' })
+    const addTeamButton = within(dialog).getByRole('button', { name: 'Continue' })
+    const inputCode = within(dialog).getByRole('textbox')
+    expect(addTeamButton).toBeDisabled()
+    expect(cancelButton).toBeVisible()
+    await userEvent.type(inputCode, '263381988')
+    expect(addTeamButton).toBeEnabled()
+    await userEvent.click(addTeamButton)
+
+    const dialogPrivacy = screen.getByRole('dialog')
+    const addCareTeamButton = within(dialogPrivacy).getByRole('button', { name: 'Join team' })
+    const checkPolicy = within(dialogPrivacy).getByRole('checkbox')
+    const textPrivatePolicy = within(dialogPrivacy).getByTestId('text-privacy-policy')
+    const detailsTeam = within(dialogPrivacy).getByTestId('team-add-dialog-team-infos')
+
+    expect(within(dialogPrivacy).getByText('You are about to share you data with')).toBeVisible()
```

Comment:
Again, you can test all the content of the dialog using `.toHaveTextContent` :slightly_smiling_face: